File: 알고리즘/Algorithm_JAVA/src/Insertion.java
public class Insertion {
    int i, j;
    int temp;
    int []a = {7, 2, 3, 9, 1};

    public void insertionSort() {
        for(i = 1; i < a.length; i++) {
            int standard = a[i];
            j = i - 1; // j는 정렬이 끝난 끝 지점을 가리킴

            while(j >= 0 && standard < a[j]) {
                a[j+1] = a[j];
                j--;
            }
            a[j+1] = standard;
        }

        for(i=0; i < a.length; i++) {
            System.out.print(a[i] + " ");
        }
    }
}

File: Algorithm/Algorithm/Algorithm/insertion.cpp
#include <iostream>

using namespace std;

void insertionSort(int arr[], int len) {
	int i, j, temp;

	for (i = 1; i < len; i++) {
		temp = arr[i];
		for (j = i - 1; j >= 0; j--) {
			if (arr[j] < temp)
				break;
			arr[j + 1] = arr[j];
		}
		arr[j + 1] = temp;
	}
}

int main() {
	int arr[6] = { 5,3,8,1,2,7 };
	
	cout << "정렬 전: ";
	for (int x : arr)
		cout << x << " ";
	cout << endl;

	insertionSort(arr, sizeof(arr) / sizeof(int));
	
	cout << "정렬 후: ";
	for (int x : arr)
		cout << x << " ";
}